docs/spec/v2alpha1/helmreleases.md

```diff
@@ -0,0 +1,276 @@
+	// +optional
+	Upgrade *struct {
+		// Timeout for the Helm upgrade action. Defaults to the main
+		// 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// Wait tells the reconciler to wait with marking the Helm upgrade as
+		// successful until all resources are in a ready state. When set, it will
+		// wait for as long as 'Timeout'. Defaults to the main 'Wait' value.
+		// +optional
+		Wait bool `json:"wait,omitempty"`
+
+		// ResetValues resets all values to the chart defaults, so that the
+		// configuration applied is what is defined in the resource values.
+		// Defaults to true to ensure declarative behaviour.
+		// +optional
+		ResetValues *bool `json:"resetValues,omitempty"`
+	} `json:"upgrade,omitempty"`
+
+	// Test holds the configuration for enabling and performing Helm
+	// test actions.
+	// +optional
+	Test *struct {
+		// Enable tells the reconciler to perform a test action
+		// on the configured condition reasons. Defaults to false.
+		// +optional
+		Enable bool `json:"enable,omitempty"`
+
+		// OnConditionReason defines on what condition reasons a Helm test
+		// should be performed. Defaults to ['InstallSucceeded', 'UpgradeSucceeded'].
+		// +optional
+		OnConditionReason []string `json:"onConditionReason,omitempty"`
+
+		// Timeout for the Helm test action. Defaults to the main
+		// 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+	} `json:"test,omitempty"`
+
+	// Rollback holds the configuration for enabling and performing Helm
+	// rollback actions.
+	// +optional
+	Rollback *struct {
+		// Enable tells the reconciler to perform a rollback action
+		// on the configured condition reasons. Defaults to false.
+		// +optional
+		Enable bool `json:"enable,omitempty"`
+		
+		// Retry tells the reconciler to retry upgrades on a rolled back release
+		// until 'MaxRetries' is reached. 
+		Retry bool `json:"retry,omitempty"`
+
+		// MaxRetries is the maximum amount of retries that should be attempted
+		// for a rolled back release. Defaults to '5' when omitted, use '0' for
+		// an unlimited amount of retries.
+		// +optional
+		MaxRetries *int `json:"maxRetries,omitempty"`
+
+		// OnConditionReason defines on what condition reasons a Helm rollback
+		// should happen. Defaults to ['UpgradeFailed', 'TestFailed'].
+		// +optional
+		OnConditionReason []string `json:"onConditionReason,omitempty"`
+
+		// Timeout for the individual operations of a Helm rollback action.
+		// Defaults to the main 'Timeout' duration.
+		// +optional
+		Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+		// DisableHooks prevents hooks from running during rollback. Defaults
+		// to false.
+		// +optional
+		DisableHooks bool `json:"disableHooks,omitempty"`
+
+		// Force resource update through delete/recreation if needed. Defaults
+		// to false.
+		// +optional
+		Force bool `json:"force,omitempty"`
```

Comment:
Does this configure the upgrade or the rollback or both? Presumably ideal would be to have this be for rollback and to add an additional `Force` field under `Upgrade` for upgrades.